x/incentives/keeper/pot_test.go

```diff
@@ -376,3 +376,48 @@ func (suite *KeeperTestSuite) TestNoLockPotDistribution() {
 	suite.Require().Len(pots, 1)
 	suite.Require().Equal(pots[0].String(), expectedPot.String())
 }
+
+func (suite *KeeperTestSuite) TestNoLockNonPerpetualPotDistribution() {
+	// test for module get pots
+	suite.SetupTest()
+
+	// setup no lock non-perpetual pot
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	potID, _, _, startTime := suite.SetupNewPot(false, coins)
+
+	// check pots
+	pots := suite.app.IncentivesKeeper.GetNotFinishedPots(suite.ctx)
+	suite.Require().Len(pots, 1)
+	expectedPot := types.Pot{
+		Id:          potID,
+		IsPerpetual: false,
+		DistributeTo: lockuptypes.QueryCondition{
+			LockQueryType: lockuptypes.ByDuration,
+			Denom:         "lptoken",
+			Duration:      time.Second,
+		},
+		Coins:             coins,
+		NumEpochsPaidOver: 2,
+		FilledEpochs:      0,
+		DistributedCoins:  sdk.Coins{},
+		StartTime:         startTime,
+	}
+	suite.Require().Equal(pots[0].String(), expectedPot.String())
+
+	// start distribution
+	suite.ctx = suite.ctx.WithBlockTime(startTime)
+	pot, err := suite.app.IncentivesKeeper.GetPotByID(suite.ctx, potID)
+	suite.Require().NoError(err)
+	err = suite.app.IncentivesKeeper.BeginDistribution(suite.ctx, *pot)
+	suite.Require().NoError(err)
+
+	// distribute coins to stakers, since it's perpetual distribute everything on single distribution
```

Comment:
stakers should be changed with bonded LPs

Reply:
I don't get the meaning of "stakers should be changed with bonded LPs"

Reply:
just the word in the comment says "stakers", I believe that it should read "bonded LPs"

Reply:
I see